detection_rules/main.py

```diff
@@ -113,6 +115,54 @@ def name_to_filename(name):
         rule_prompt(rule_path, required_only=True, save=True, verbose=True, additional_required=['index'], **contents)
 
 
+@root.command('build-limited-rules')
+@click.option('--stack-version', type=click.Choice(all_versions()), required=True,
+              help='Version to downgrade to be compatible with the older instance of Kibana')
+@click.option('--output-file', '-o', type=click.Path(dir_okay=False, exists=False), required=True)
+def build_limited_rules(stack_version: str, output_file: str):
+    """
+    Import rules from json, toml, or Kibana exported rule file(s),
+    filter out unsupported ones, and write to output NDJSON file.
+    """
+
+    # Schema generation and incompatible fields detection
+    query_rule_data = class_schema(QueryRuleData)()
+    fields = getattr(query_rule_data, 'fields', {})
+    incompatible_fields = get_incompatible_fields(list(fields.values()),
+                                                  Version.parse(stack_version, optional_minor_and_patch=True))
+
+    # Load all rules
+    rules = RuleCollection.default()
+
+    # Define output path
+    output_path = Path(output_file)
+
+    # Function to process and write each rule to the output file
+    def process_and_write_rule(rule, incompatible_fields: List[str], file):
+        if rule.contents.type in UNSUPPORTED_RULE_TYPES:
+            click.secho(f'Skipping unsupported rule type: {rule.contents.get("type")}', fg='yellow')
+            return
+
+        # Remove unsupported fields from rule
+        rule_contents = rule.contents.to_api_format()
+        for field in incompatible_fields:
+            rule_contents.pop(field, None)
+
+        # Try to write the cleaned rule to the NDJSON output file
+        try:
+            json.dump(rule_contents, file)
```

Comment:
Should this use `utils.py`'s ndjson class .dump() ?

Reply:
Which function are you thinking? I think the dump will write one at a time, whereas here we're iterating over multiple files writing to the same ndjson.